dashboard/modules/data/data_head.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os
+from enum import Enum
+import aiohttp
+from aiohttp.web import Request, Response
+import ray.dashboard.optional_utils as optional_utils
+import ray.dashboard.utils as dashboard_utils
+from ray.data._internal.stats import _get_or_create_stats_actor
+from ray.dashboard.modules.metrics.metrics_head import (
+    PROMETHEUS_HOST_ENV_VAR,
+    DEFAULT_PROMETHEUS_HOST,
+    PrometheusQueryError,
+)
+from urllib.parse import quote
+import ray
+
+
+MAX_TIME_WINDOW = "1h"
+SAMPLE_RATE = "1s"
+
+
+class PrometheusQuery(Enum):
+    VALUE = ("value", "sum({}) by (dataset)")
+    MAX = (
+        "max",
+        "max_over_time(sum({}) by (dataset)[" + f"{MAX_TIME_WINDOW}:{SAMPLE_RATE}])",
+    )
+
+
+DATASET_METRICS = {
+    "ray_data_output_bytes": (PrometheusQuery.MAX,),
+    "ray_data_spilled_bytes": (PrometheusQuery.MAX,),
+    "ray_data_current_bytes": (PrometheusQuery.VALUE, PrometheusQuery.MAX),
+}
+
+
+class DataHead(dashboard_utils.DashboardHeadModule):
+    def __init__(self, dashboard_head):
+        super().__init__(dashboard_head)
+        self.http_session = aiohttp.ClientSession()
+        self.prometheus_host = os.environ.get(
+            PROMETHEUS_HOST_ENV_VAR, DEFAULT_PROMETHEUS_HOST
+        )
+
+    @optional_utils.DashboardHeadRouteTable.get("/api/data/datasets")
+    @optional_utils.init_ray_and_catch_exceptions()
+    async def get_datasets(self, req: Request) -> Response:
+        try:
+            _stats_actor = _get_or_create_stats_actor()
+            datasets = ray.get(_stats_actor.get_datasets.remote())
+            # Initializes dataset metric values
+            for dataset in datasets:
+                for metric, queries in DATASET_METRICS.items():
+                    datasets[dataset][metric] = {query.value[0]: 0 for query in queries}
+            # Query dataset metric values from prometheus
+            for metric, queries in DATASET_METRICS.items():
+                for query in queries:
+                    result = await self._query_prometheus(query.value[1].format(metric))
+                    for res in result["data"]["result"]:
+                        dataset, value = res["metric"]["dataset"], res["value"][1]
+                        if dataset in datasets:
```

Comment:
is there any case where this would not be true? if so, would we want to also include the dataset in the results?

Reply:
If you don't restart the prometheus server, this can have extra datasets across different jobs